test/GSN/ERC721GSNRecipientMock.test.js

```diff
@@ -0,0 +1,46 @@
+const { constants, expectEvent } = require('openzeppelin-test-helpers');
+const { ZERO_ADDRESS } = constants;
+const gsn = require('@openzeppelin/gsn-helpers');
+const { fixSignature } = require('../helpers/sign');
+const { utils: { toBN } } = require('web3');
+
+const ERC721GSNRecipientMock = artifacts.require('ERC721GSNRecipientMock');
+
+contract('ERC721GSNRecipient', function ([_, signer, sender]) {
```

Comment:
We should signal that this is an integration test. How about the following?

```suggestion
contract('ERC721GSNRecipient (integration)', function ([_, signer, sender]) {
```